nomad/grapl-core.nomad

```diff
@@ -0,0 +1,428 @@
+            network {
+                mode = "bridge"
+            }
+
+            task "dgraph-alpha" {
+                driver = "docker"
+
+                config {
+                    image = "dgraph/dgraph:latest"
+                    args = [
+                        "dgraph",
+                        "alpha",
+                        "--my=localhost:${alpha.value.grpc_private_port}",
+                        "-o", "${alpha.value.id}",
+                        "--zero", "${local.zero_alpha_connect_str}"
+                    ]
+                }
+            }
+
+            service {
+                name = "dgraph-alpha-${alpha.value.id}-grpc-private"
+                port = "${alpha.value.grpc_private_port}"
+
+                connect {
+                    sidecar_service {
+                        proxy {
+                            # Connect to the Zero leader
+                            upstreams {
+                                destination_name = "dgraph-zero-leader-grpc-private"
+                                local_bind_port = 5080
+                            }
+
+                            # Connect this Alpha to Zero followers
+                            dynamic "upstreams" {
+                                iterator = zero_follower
+                                for_each = [for zero_follower in local.dgraph_zeros: zero_follower]
+
+                                content {
+                                    destination_name = "dgraph-zero-follower-${zero_follower.value.id}-grpc-private"
+                                    local_bind_port = zero_follower.value.grpc_private_port
+                                }
+                            }
+
+                            # Connect this Alpha to Other Alphas (but not to itself, obviously)
+                            dynamic "upstreams" {
+                                iterator = alpha_peer
+                                for_each = [for alpha_peer in local.dgraph_alphas: alpha_peer if alpha_peer.id != alpha.value.id]
+
+                                content {
+                                    destination_name = "dgraph-alpha-${alpha_peer.value.id}-grpc-private"
+                                    local_bind_port = alpha_peer.value.grpc_private_port
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            service {
+                name = "dgraph-alpha-${alpha.value.id}-grpc-public"
+                port = "${alpha.value.grpc_public_port}"
+
+                connect {
+                    sidecar_service { }
+                }
+            }
+
+            service {
+                name = "dgraph-alpha-${alpha.value.id}-http"
+                port = "${alpha.value.http_port}"
+
+                connect {
+                    sidecar_service { }
+                }
+            }
+        }
+    }
+
+    group "grapl-graph-merger" {
```

Comment:
(Doesn't have to be done during this PR)
Are we able to split up the different `group`s at all or do they all have to be in 1 file?

Reply:
As far as I'm aware, it must all be defined together :/